crates/core/src/rpc/full.rs

```diff
@@ -1956,11 +1956,61 @@ impl Full for SurfpoolFullRpc {
 
     fn get_signatures_for_address(
         &self,
-        _meta: Self::Metadata,
-        _address: String,
+        meta: Self::Metadata,
+        address: String,
         _config: Option<RpcSignaturesForAddressConfig>,
     ) -> BoxFuture<Result<Vec<RpcConfirmedTransactionStatusWithSignature>>> {
-        not_implemented_err_async("get_signatures_for_address")
+        let address = match verify_pubkey(&address) {
+            Ok(s) => s,
+            Err(e) => return e.into(),
+        };
+        let SurfnetRpcContext {
+            svm_locker,
+            remote_ctx,
+        } = match meta.get_rpc_context(None) {
+            Ok(res) => res,
+            Err(e) => return e.into(),
+        };
+
+        Box::pin(async move {
+            let sigs: Vec<_> = svm_locker.with_svm_reader(|svm_reader| {
+                svm_reader
+                    .transactions
+                    .iter()
+                    .filter(|(_, status)| {
+                        let TransactionWithStatusMeta(_, tx, _, _) = status.expect_processed();
+
+                        tx.message
+                            .static_account_keys()
+                            .iter()
+                            .find_position(|pubkey| pubkey.to_bytes() == address.to_bytes())
+                            .map(|(idx, _)| tx.message.is_signer(idx))
+                            .unwrap_or_default()
+                    })
+                    .map(|(sig, status)| (sig.to_owned(), status.to_owned()))
+                    .collect()
+            });
+
+            let mut responses = Vec::with_capacity(sigs.len());
+
+            for (signature, status) in sigs.into_iter() {
+                let res = svm_locker.get_transaction(&remote_ctx, &signature).await;
```

Comment:
Looks like we only needed to fetch the transaction to find the confirmation status. I think we can skip this fetch, we've been using the following to get the status:
 - `slot == svm_locker.get_latest_block_hash()` - Processed
 - `slot == svm_locker.get_latest_block_hash().saturating_sub(1)` - Confirmed
 - `slot == svm_locker.get_latest_block_hash().saturating_sub(FINALIZATION_SLOT_THRESHOLD)` - Finalized